server/task_check_scheduler.go

```diff
@@ -179,6 +179,74 @@ func (s *TaskCheckScheduler) Register(taskType string, executor TaskCheckExecuto
 
 // ScheduleCheckIfNeeded schedules a check if needed.
 func (s *TaskCheckScheduler) ScheduleCheckIfNeeded(ctx context.Context, task *api.Task, creatorID int, skipIfAlreadyTerminated bool) (*api.Task, error) {
+	// All task should pass timing task check.
+	// Since Time is an auto-increment value, a task would generate a timing task check every time before executing.
+	// However, this is an annoying behavior, and we adopt the following logic to ameliorate this problem.
+	{
+		// fetching history task check
+		var err error
+		var timingTaskCheck *api.TaskCheckRun
+		task.TaskCheckRunList, err = s.server.TaskCheckRunService.FindTaskCheckRunList(ctx, &api.TaskCheckRunFind{TaskID: &task.ID})
+		if err != nil {
+			return nil, err
+		}
+		for _, taskCheck := range task.TaskCheckRunList {
+			if taskCheck.Type == api.TaskCheckGeneralEarliestAllowedTime {
+				timingTaskCheck = taskCheck
+				break
+			}
+		}
+		isPassedTimingTaskCheck, err := s.server.passCheck(ctx, s.server, task, api.TaskCheckGeneralEarliestAllowedTime)
```

Comment:
should this put inside the else block below instead of putting it before if timingTaskCheck == nil block?